oximeter/db/src/client.rs

```diff
@@ -3338,4 +3680,524 @@ mod tests {
             );
         }
     }
+
+    async fn create_test_upgrade_schema_directory(
+        replicated: bool,
+        versions: &[u64],
+    ) -> (TempDir, Vec<PathBuf>) {
+        assert!(!versions.is_empty());
+        let schema_dir = TempDir::new().expect("failed to create tempdir");
+        let mut paths = Vec::with_capacity(versions.len());
+        for version in versions.iter() {
+            let version_dir = Client::full_upgrade_path(
+                replicated,
+                *version,
+                schema_dir.as_ref(),
+            );
+            fs::create_dir_all(&version_dir)
+                .await
+                .expect("failed to make version directory");
+            paths.push(version_dir);
+        }
+        (schema_dir, paths)
+    }
+
+    #[tokio::test]
+    async fn test_read_schema_upgrade_sql_files() {
+        let logctx = test_setup_log("test_read_schema_upgrade_sql_files");
+        let log = &logctx.log;
+        const REPLICATED: bool = false;
```

Comment:
Let's test for replicated as well

Reply:
This test doesn't actually require a database, it's just using this to find the right schema subdirectory.

Reply:
Yes, we want to make sure it also finds the replicated schema subdirectory as well.